pkg/appliance/functions.go

```diff
@@ -596,6 +584,134 @@ func applyApplianceFilter(appliances []openapi.Appliance, filter map[string]stri
 	return filteredAppliances, nil
 }
 
+func orderAppliances(appliances []openapi.Appliance, orderBy []string, descending bool) ([]openapi.Appliance, error) {
+	// reverse loop the slice to prioritize the ordering. First entered has priority
+	for i := len(orderBy) - 1; i >= 0; i-- {
+		switch orderBy[i] {
+		case "name":
+			if descending {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetName() > appliances[j].GetName() })
+			} else {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetName() < appliances[j].GetName() })
+			}
+		case "id":
+			if descending {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetId() > appliances[j].GetId() })
+			} else {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetId() < appliances[j].GetId() })
+			}
+		case "site":
+			if descending {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetSite() > appliances[j].GetSite() })
+			} else {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetSite() < appliances[j].GetSite() })
+			}
+		case "site-name":
+			if descending {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetSiteName() > appliances[j].GetSiteName() })
+			} else {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetSiteName() < appliances[j].GetSiteName() })
+			}
+		case "hostname", "host":
+			if descending {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetHostname() > appliances[j].GetHostname() })
+			} else {
+				sort.SliceStable(appliances, func(i, j int) bool { return appliances[i].GetHostname() < appliances[j].GetHostname() })
+			}
+		case "activated", "active":
+			if descending {
+				sort.SliceStable(appliances, func(i, j int) bool {
+					return !appliances[i].GetActivated() && appliances[i].GetActivated() != appliances[j].GetActivated()
+				})
+			} else {
+				sort.SliceStable(appliances, func(i, j int) bool {
+					return appliances[i].GetActivated() && appliances[i].GetActivated() != appliances[j].GetActivated()
+				})
+			}
+		default:
+			log.WithField("keyword", orderBy[i]).Warn("keyword not sortable, ignoring")
```

Comment:
redundant log, if you really want log for these, maybe we should demote it to DEBUG?

Reply:
I wouldn't call any of these redundant. If I typo one of the kywords and don't understand why the ordering isn't the way it's supposed to be, I wouldn't want to need to set up DEBUG mode just to figure out that I mistyped something.

Reply:
In that case I think the command should fail with a non zero exit code immediately. No point in running at all if the command can't do what we tell it to do.